mint-auditor/src/db/mod.rs

```diff
@@ -1024,13 +1079,12 @@ mod tests {
         let block =
             Block::new_with_parent(BLOCK_VERSION, &block, &Default::default(), &block_contents);
 
-        mint_audit_db
-            .sync_block(&block, &block_contents, &ledger_db)
-            .unwrap();
         ledger_db
             .append_block(&block, &block_contents, None)
             .unwrap();
 
+        mint_audit_db.sync_block(&block, &block_contents).unwrap();
```

Comment:
Will do in a followup PR since this one already has a lot of changes.

Reply:
I started pulling on this thread and to add the stuff you did in #2052 would require duplicating a bunch of the work and I'm not sure its worth it instead of waiting for #2052 to go in and then use that.